### PR TITLE
clientv3/integration: add timeouts to TestTxnWriteFail

### LINKED ISSUE #4424: test: TestTxnWriteFail fails
```
=== RUN TestTxnWriteFail
...
timeout after 10minutes
```

### PR DESCRIPTION
**Pull request by heyitsanthony**
also made TestTxnWriteFail better about timing out

fixes #4424

### CODE REVIEW

File: clientv3/client.go [hunk 1]
```diff
@@ -200,5 +200,6 @@ func dialEndpointList(c *Client) (*grpc.ClientConn, error) {
 }
 
 func isRPCError(err error) bool {
-	return grpc.Code(err) != codes.Unknown
+	return grpc.Code(err) != codes.Unknown &&
```
Comment (xiang90) [inline clientv3/client.go]: the commit message says `treat closing client as RPC error`.

But here we treat grpc.ErrClientConnClosing as !RPCError. I think the code is correct.

Comment (heyitsanthony) [inline clientv3/client.go]: grpc is giving me code = 2 = codes.Unknown for it

**Comment by xiang90** [comment]
@heyitsanthony Are you able to reproduce the timeout issue reliably?

**Comment by heyitsanthony** [comment]
@xiang90 yes, after looping the test a few times

File: clientv3/client.go (hunk 1 above)
Comment (xiang90) [inline clientv3/client.go]: > grpc is giving me code = 2 = codes.Unknown for it

Yea... If that is the case, this line of change does not change any logic.

Comment (heyitsanthony) [inline clientv3/client.go]: `rpc error: code = 2 desc = "grpc: the client connection is closing"` => isRPCError() is false. I think I just screwed up the conditional when writing the fix.

**Comment by heyitsanthony** [comment]
all fixed ptal /cc @xiang90

File: clientv3/client.go [hunk 2]
```diff
@@ -200,5 +200,6 @@ func dialEndpointList(c *Client) (*grpc.ClientConn, error) {
 }
 
 func isRPCError(err error) bool {
-	return grpc.Code(err) != codes.Unknown
+	return grpc.Code(err) != codes.Unknown ||
+		err == grpc.ErrClientConnClosing
```
Comment (xiang90) [inline clientv3/client.go]: If the current client conn is closing, should'nt we retry? (when we meet rpc error, we will not retry)

We are not sure about if just the conn is closing or the entire client is closing, right? I feel we should only exit when the client ctx is cancelled.

Comment (heyitsanthony) [inline clientv3/client.go]: OK, it looks like it's timing out on Get and the change breaks out of it but strangely Get isn't returning an error like it should.

**Comment by heyitsanthony** [comment]
timeout seems to be happening in Get(), will investigate kv but might as well have this test fail fast in the meantime

**Comment by xiang90** [comment]
lgtm.

**Pull request merged**